app/models/framework_response.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class FrameworkResponse < VersionedModel
+  validates :type, presence: true
+
+  belongs_to :framework_question
+  belongs_to :person_escort_record
+  has_many :dependents, class_name: 'FrameworkResponse',
+                        foreign_key: 'parent_id'
+
+  belongs_to :parent, class_name: 'FrameworkResponse', optional: true
+  validates_each :value, on: :update do |record, _attr, value|
+    if value.blank? && record.parent&.option_selected?(record.framework_question.dependent_value) && record.framework_question.required
+      record.errors.add(:value, :blank)
+    end
+  end
+
+  def self.find_sti_class(type_name)
```

Comment:
Since this is implicitly overriding the method that lives in `ActiveRecord::Inheritance` I'd add a comment to save people time in understanding this.

`NB: Overrides mechanism used by ActiveRecord to instantiate objects in STI` or equivalent

Reply:
This implementation has now been removed